File: RepositorioEjercicios/Java/java-gradle-testing/src/main/java/org/example/N_30_AdvancedAlgorithms/MergeTwoSortedLists.java
package org.example.N_30_AdvancedAlgorithms;

import java.util.Arrays;
import java.util.List;

public class MergeTwoSortedLists {
    public static int[] mergeTwoSortedLists(int[] list1, int m, int[] list2, int n) {
        int pointerOne = m+n-1;
        int pointerTwo = m-1;
        int pointerThree = n-1;
        while (pointerOne >= 0 && pointerTwo >= 0 && pointerThree >= 0){
            if (list2[pointerThree] > list1[pointerTwo]){
                list1[pointerOne] = list2[pointerThree];
                pointerThree--;
            } else if (list2[pointerThree] < list1[pointerTwo]) {
                list1[pointerOne] = list1[pointerTwo];
                pointerTwo--;
            } else if (list2[pointerThree] == list1[pointerTwo]) {
                list1[pointerOne] = list2[pointerThree];
                pointerTwo--;
            }
            pointerOne--;
        }
        if (pointerOne >= 0 && pointerThree >= 0){
            list1[pointerOne] = list2[pointerThree];
        }
        System.out.println(Arrays.toString(list1));
        return list1;
    }

    public static void main(String[] args) {
        int[] list1 = {1,2,3,0,0,0};
        int m1 = 3;
        int n1 = 3;
        int[] list2 = {2,5,6};
        mergeTwoSortedLists(list1,m1,  list2, n1);
        int[] nums1 = {1,2,3,0,0,0,0};
        int m = 3;
        int[] nums2 = {-4,2,3,9};
        int n = 4;
        mergeTwoSortedLists(nums1, m, nums2, n);
    }
}

File: RepositorioEjercicios/Python/pythoninicial/src/LeetCode/MergeTwoSortedLists.py
def merge_lists(nums1, m, nums2, n):
    # Tu código aquí 👇
    pointerOne = m + n - 1
    pointerTwo = m - 1
    pointerThree = n - 1

    while pointerOne >= 0 and pointerTwo >= 0 and pointerThree >= 0:
        if nums2[pointerThree] > nums1[pointerTwo]:
            nums1[pointerOne] = nums2[pointerThree]
            pointerThree -= 1
        elif nums2[pointerThree] < nums1[pointerTwo]:
            nums1[pointerOne] = nums1[pointerTwo]
            pointerTwo -= 1
        elif nums2[pointerThree] == nums1[pointerTwo]:
            nums1[pointerOne] = nums2[pointerThree]
            pointerTwo -= 1
        pointerOne -= 1
    if pointerOne >= 0 and pointerThree >= 0:
        nums1[pointerOne] = nums2[pointerThree]
    return nums1


# Arrays de prueba 1
nums1 = [1, 2, 3, 0, 0, 0]
m = 3
nums2 = [2, 5, 6]
n = 3
response = merge_lists(nums1, m, nums2, n)
print(response)

# Arrays de prueba 2
nums1 = [1, 2, 3, 0, 0, 0, 0]
m = 3
nums2 = [-4, 2, 3, 9]
n = 4
response = merge_lists(nums1, m, nums2, n)
print(response)
